From a57bc72be59974faa01ed46cf565bd527e8a45bb Mon Sep 17 00:00:00 2001
From: Ulf Hermann <ulf.hermann@qt.io>
Date: Wed, 17 Apr 2019 12:35:42 +0200
Subject: [PATCH 4/5] Don't crash when accessing invalid properties through
 QObjectWrapper

Change-Id: I613bf5dc685bb4235262b429d8f7318ea144fb9d
Fixes: QTBUG-75203
Reviewed-by: Erik Verbruggen <erik.verbruggen@qt.io>

Conflicts:
    src/qml/jsruntime/qv4qobjectwrapper.cpp
    tests/auto/qml/qqmlecmascript/tst_qqmlecmascript.cpp
    tests/auto/qml/qqmlecmascript/data/undefinedPropertiesInObjectWrapper.qml

(cherry picked from commit ea74f0c68cddf706c950d3910cf7b363fe24885b)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/qml/jsruntime/qv4qobjectwrapper.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/qml/jsruntime/qv4qobjectwrapper.cpp b/src/qml/jsruntime/qv4qobjectwrapper.cpp
index 326381f38d..b1bd64772b 100644
--- a/src/qml/jsruntime/qv4qobjectwrapper.cpp
+++ b/src/qml/jsruntime/qv4qobjectwrapper.cpp
@@ -281,6 +281,9 @@ ReturnedValue QObjectWrapper::getQmlProperty(QQmlContextData *qmlContext, String
 
 ReturnedValue QObjectWrapper::getProperty(ExecutionEngine *engine, QObject *object, QQmlPropertyData *property, bool captureRequired)
 {
+    if (!property)
+        return QV4::Encode::undefined();
+
     QQmlData::flushPendingBinding(object, QQmlPropertyIndex(property->coreIndex()));
 
     if (property->isFunction() && !property->isVarProperty()) {
-- 
2.20.1

